app/views/TaggingFlow/PrimaryTagging/index.tsx

```diff
@@ -23,6 +26,41 @@ function PrimaryTagging(props: Props) {
         ]));
     }, [setEntries]);
 
+    const handleEntryDelete = React.useCallback((entryId: string) => {
+        setEntries((oldEntries) => {
+            const i = oldEntries.findIndex((e) => e.clientId === entryId);
+
+            if (i === -1) {
+                return oldEntries;
+            }
+
+            const newEntries = [...oldEntries];
+            newEntries.splice(i, 1);
+
+            return newEntries;
+        });
+    }, [setEntries]);
+
+    const handleExcerptChange = React.useCallback((entryId: string, modifiedExcerpt: string) => {
+        setEntries((oldEntries) => {
+            const i = oldEntries.findIndex((e) => e.clientId === entryId);
+
+            if (i === -1) {
+                return oldEntries;
+            }
+
+            const newEntries = [...oldEntries];
+            const newEntry = {
+                ...(newEntries[i]),
+                excerpt: modifiedExcerpt,
```

Comment:
We should also set pristine/non-prisinte value

Reply:
That can be simply calculated by comparing `droppedExcerpt` and `excerpt`